### PR TITLE
[Example] Add Pytorch Geometric Example

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds Pytorch Geometric Examples with Lightning.

Fixes # (issue)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tchaton** [comment]
Screenshot 2020-11-07 at 15 17 33

File: pl_examples/pytorch_ecosystem/pytorch_geometric/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+pyenv install 3.7.8
+pyenv local 3.7.8
+python -m venv
+source .venv/bin/activate
+poetry install
+```
+
+## Current example lists
+
+| `DATASET` | `MODEL` | `TASK` | DATASET DESCRIPTION | MODEL DESCRIPTION                                                                                                                                                                   |                                                                                                                                                                     |
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/README.md:23]: @ananyahjha93 used another type of writing table wich turned to be much easier to read in the source
https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table

File: pl_examples/pytorch_ecosystem/pytorch_geometric/poetry.lock [hunk 2]
```diff
@@ -0,0 +1,1637 @@
+[[package]]
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/poetry.lock:1]: i don't think that this super long lock file is needed... :]

Comment (tchaton) [reply pl_examples/pytorch_ecosystem/pytorch_geometric/poetry.lock:1]: Yes, you are right ! I was going to remove it.

File: pl_examples/pytorch_ecosystem/pytorch_geometric/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+[tool.poetry]
```
Comment (SeanNaren) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/pyproject.toml:1]: Is this file needed for the example? The less files the better imo

Comment (tchaton) [reply pl_examples/pytorch_ecosystem/pytorch_geometric/pyproject.toml:1]: Install with Pytorch Geometric can be pretty complicated. `poetry install` makes it simpler.

**Review by SeanNaren — APPROVED** [review]
I'm hoping that lightning logo will make its way into the trainer, would be neat :)

File: pl_examples/pytorch_ecosystem/pytorch_geometric/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Run example
+
+```
+python cora_dna.py
+```
+
+## Current example lists
+
+| `DATASET` | `MODEL` | `TASK` | DATASET DESCRIPTION | MODEL DESCRIPTION                                                                                                                                                                   |                                                                                                                                                                     |
+| ------------- | ----------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------------------------- | --- |
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/README.md:30]:
```suggestion
| :---: | :---: | :---: | :---: | :---: | :---: |
```

File: pl_examples/pytorch_ecosystem/pytorch_geometric/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+| Cora          | DNA          | Node Classification | The citation network datasets "Cora", "CiteSeer" and "PubMed" from the "Revisiting Semi-Supervised Learning with Graph Embeddings" <https://arxiv.org/abs/1603.08861> | The dynamic neighborhood aggregation operator from the "Just Jump: Towards Dynamic Neighborhood Aggregation in Graph Neural Networks"
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/README.md:31]:
```suggestion
| Cora | DNA | Node Classification | The citation network datasets "Cora", "CiteSeer" and "PubMed" from the "Revisiting Semi-Supervised Learning with Graph Embeddings" <https://arxiv.org/abs/1603.08861> | The dynamic neighborhood aggregation operator from the "Just Jump: Towards Dynamic Neighborhood Aggregation in Graph Neural Networks"
```

File: pl_examples/pytorch_ecosystem/pytorch_geometric/cora_dna.py [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+    # Pytorch Geometric imports
+    from torch_geometric.nn import DNAConv, MessagePassing
+    from torch_geometric.data import DataLoader
+    from torch_geometric.datasets import Planetoid
+    import torch_geometric.transforms as T
+    from torch_geometric.data import NeighborSampler
+    from lightning import lightning_logo, nice_print
+    HAS_PYTORCH_GEOMETRIC = True
+except Exception:
+    HAS_PYTORCH_GEOMETRIC = False
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/cora_dna.py:42]:
```suggestion
except Exception:
    HAS_PYTORCH_GEOMETRIC = False
else:
    HAS_PYTORCH_GEOMETRIC = True
```

**Review by Borda — COMMENTED** [review]
minor formating

**Review by ananyahjha93 — APPROVED** [review]

**Review by teddykoker — APPROVED** [review]
LGTM, cool example

File: pl_examples/pytorch_ecosystem/pytorch_geometric/lightning.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
+def nice_print(msg, last=False):
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/lightning.py:1]: Let's move this to PL_examples root?

File: pl_examples/pytorch_ecosystem/pytorch_geometric/pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+name = "lightning-geometric"
+version = "0.1.0"
+description = "TorchScripted Pytorch Geometric Examples with Pytorch Lightning"
+authors = ["Thomas Chaton <thomas.ai@grid.com>"]
```
Comment (Borda) [inline pl_examples/pytorch_ecosystem/pytorch_geometric/pyproject.toml:5]: still the mail here...
```suggestion
authors = ["Thomas Chaton <thomas@pytorchlightning.ai>"]
```

**Review by Borda — COMMENTED** [review]
can we pls add these: @tchaton 
- [ ] needed packages to `requirements/examples.txt`
- [ ] test run, even GPU only, how large is the dataset?

**Comment by Borda** [comment]
some fixes come in #5085
